tests/schema/test_private_field.py

```diff
@@ -55,3 +55,29 @@ def age_in_months(self) -> int:
     assert result.data == {
         "ageInMonths": 84,
     }
+
+
+def test_private_field_with_str_annotations():
+    """Check compatibility of strawberry.Private with annotations as string."""
+
+    from dataclasses import dataclass
+
+    @strawberry.type
+    class Query:
+        not_seen: "strawberry.Private[SensitiveData]"
+
+        @strawberry.field
+        def accesible_info(self) -> str:
+            return self.not_seen.info
+
+    @dataclass
+    class SensitiveData:
+        value: int
+        info: str
+
+    schema = strawberry.Schema(query=Query)
+
+    result = schema.execute_sync(
+        "query { accesibleInfo }", root_value=Query(not_seen=SensitiveData(1, "foo"))
+    )
+    assert result.data == {"accesibleInfo": "foo"}
```

Comment:
Should we be instead (or in addition) having this test ensure that `not_seen` does not exist on the schema? This test passes exactly the same if the `strawberry.Private` wrapping is simply removed.

Also, a couple nitpicky requests:
* Unless it serves a purpose for this test (which I don't think it does :thinking:), can you move the `dataclass` import to the top of the file?
* `accessible` is spelled wrong

Reply:
Hey @BryceBeagle thanks a lot for the review! You made a good point about the test still passing if `strawberry.Private` is removed. I'll fix that right away